File: app/src/main/java/com/cna/mineru/cna/Utils/SecurityUtil.java
package com.cna.mineru.cna.Utils;

import android.util.Base64;

import java.nio.charset.Charset;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class SecurityUtil {
    public byte[] encryptSHA256(String str){
        try{
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] byteData = str.getBytes(Charset.forName("UTF-8"));
            md.update(byteData);
            return Base64.encode(md.digest(),0);
        }catch (NoSuchAlgorithmException e){
            return null;
        }
    }
}

File: app/src/main/java/com/cna/mineru/cna/Utils/SecurityUtil.kt
package com.cna.mineru.cna.Utils

import android.util.Base64

import java.nio.charset.Charset
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

class SecurityUtil {
    fun encryptSHA256(str: String): ByteArray? {
        try {
            val md = MessageDigest.getInstance("SHA-256")
            val byteData = str.toByteArray(Charset.forName("UTF-8"))
            md.update(byteData)
            return Base64.encode(md.digest(), 0)
        } catch (e: NoSuchAlgorithmException) {
            return null
        }

    }
}